src/main/java/org/gridsuite/directory/server/DirectoryController.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.gridsuite.directory.server;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import org.gridsuite.directory.server.dto.AccessRightsAttributes;
+import org.gridsuite.directory.server.dto.CreateDirectoryAttributes;
+import org.gridsuite.directory.server.dto.ElementAttributes;
+import org.gridsuite.directory.server.repository.DirectoryElementEntity;
+import org.gridsuite.directory.server.repository.DirectoryRootEntity;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+/**
+ * @author Nicolas Noir <nicolas.noir at rte-france.com>
+ */
+@RestController
+@RequestMapping(value = "/" + DirectoryApi.API_VERSION)
+@Api(tags = "directory-server")
+public class DirectoryController {
+
+    private final DirectoryService service;
+
+    public DirectoryController(DirectoryService service) {
+        this.service = service;
+    }
+
+    @GetMapping(value = "/directories/root", produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiOperation(value = "Get root directory id")
+    @ApiResponses(@ApiResponse(code = 200, message = "Successfully get root directory id"))
+    public ResponseEntity<Mono<DirectoryRootEntity>> getRootDirectoryId(@RequestHeader("userId") String headerUserId) {
+        return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(service.getRootDirectory());
+    }
+
+    @PostMapping(value = "/directories/create")
+    @ApiOperation(value = "Create directory")
+    @ApiResponses(@ApiResponse(code = 200, message = "Successfully created directory"))
+    public ResponseEntity<Mono<DirectoryElementEntity>> createDirectory(@RequestBody CreateDirectoryAttributes createDirectoryAttributes,
+                                                                        @RequestHeader("userId") String headerUserId) {
```

Comment:
headerUserId parameters are never used. Is this in purpose ?

Reply:
This parameter was supposed to be used to check user's access rights. Since access rights check is not implemented in this version, the parameter has been removed.